protocols/raft/src/main/java/io/atomix/protocols/raft/impl/RaftServiceManager.java

```diff
@@ -494,6 +501,7 @@ private void compactLog() {
           .whenCompleteAsync((result, error) -> {
             logger.info("Compacting logs up to index {}", lastApplied);
             log.compact(lastApplied);
+            scheduleCompaction();
```

Comment:
Can log.compact ever throw? Want to make sure we don't miss the next `scheduleCompaction`.

Reply:
I suppose it could, due to an `IOException` in the log. Should probably log that as an `ERROR`